monkey/infection_monkey/exploit/hadoop.py

```diff
@@ -40,11 +46,25 @@ def _exploit_host(self):
         urls = self.build_potential_urls(self.host.ip_addr, self.HADOOP_PORTS)
         self.add_vulnerable_urls(urls, True)
         if not self.vulnerable_urls:
+            self.exploit.error_message = f"No vulnerable urls has been found for {self.host}"
+            self._publish_exploitation_event(
+                target=self.host.ip_addr,
+                exploitation_success=False,
+                error_message=self.exploit_result.error_message,
+                tags=(HADOOP_EXPLOITER_TAG,),
+            )
             return self.exploit_result
 
         try:
             monkey_path_on_victim = get_agent_dst_path(self.host)
         except KeyError:
+            self.exploit_result.error_message = f"No coressponding agent found for {self.host}"
+            self._publish_exploitation_event(
+                target=self.host.ip_addr,
+                exploitation_success=False,
+                error_message=self.exploit_result.error_message,
+                tags=(HADOOP_EXPLOITER_TAG,),
+            )
```

Comment:
I don't see how `get_agent_dst_path()` can ever raise a `KeyError`. This may be left over. See if I missed something and then fix :)